packages/fuels-core/src/types/checksum_address.rs

```diff
@@ -0,0 +1,138 @@
+use crate::types::errors::{Error, Result};
+use sha3::{Digest, Keccak256};
+
+pub fn checksum_encode(address: &str) -> Result<String> {
+    let trimmed = address.trim_start_matches("0x");
+    pre_validate(trimmed)?;
+
+    let lowercase = trimmed.to_ascii_lowercase();
+
+    let hash = Keccak256::digest(lowercase.as_bytes());
```

Comment:
We're not using Keccak on the TS side - can you confirm the test cases we have in https://github.com/FuelLabs/fuels-ts/pull/3308/files#diff-2c4d8f2d738f195a78c5ee15cb8b4271401b169fefbdccb0e1924b33e74b1cabR381-R400 all pass the same way here?

Reply:
I'll switch it over to sha256 if that's what we want to do